### PR TITLE
libcgroup: 0.41 -> 0.42.2

### PR DESCRIPTION
**Pull request by LeSuisse**
###### Motivation for this change

Avoiding false positive when checking for security issues (#99797) and update the package...

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>afl</li>
    <li>aflplusplus</li>
    <li>libcgroup</li>
  </ul>
</details>

### CODE REVIEW

File: pkgs/os-specific/linux/libcgroup/default.nix [hunk 1]
```diff
@@ -1,26 +1,21 @@
...
 stdenv.mkDerivation rec {
   pname = "libcgroup";
-  version = "0.41";
+  version = "0.42.2";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/libcg/${pname}-${version}.tar.bz2";
-    sha256 = "0lgvyq37gq84sk30sg18admxaj0j0p5dq3bl6g74a1ppgvf8pqz4";
+  src = fetchFromGitHub {
+    owner = pname;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libcgroup/default.nix:8]: I am not sure if this is credible. There website does not link it.

Comment (LeSuisse) [reply pkgs/os-specific/linux/libcgroup/default.nix:8]: It's what the [Fedora project is currently using](https://src.fedoraproject.org/rpms/libcgroup/blob/17704306e5f5e3a42edd3254f414c72996a5b138/f/libcgroup.spec#_7), the [mailing list libcg-devel](https://sourceforge.net/p/libcg/mailman/libcg-devel/) seems to match what's in the Git repository on GitHub and there are messages about features specific to GitHub (see this thread for example: https://sourceforge.net/p/libcg/mailman/libcg-devel/thread/20201130171650.14825-3-tom.hromatka%40oracle.com/#msg37165452).

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110560` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>afl</li>
<li>aflplusplus</li>
<li>libcgroup</li>
</ul>
</details>

File: pkgs/os-specific/linux/libcgroup/default.nix (hunk 1 above)
Comment (LeSuisse) [reply pkgs/os-specific/linux/libcgroup/default.nix:8]: FYI the move to GitHub is now mentioned on the SourceForge page.

https://sourceforge.net/projects/libcg/

**Pull request merged**